src/optimizer/remove_unused_columns.cpp

```diff
@@ -57,13 +57,14 @@ void RemoveUnusedColumns::VisitOperator(LogicalOperator &op) {
 			auto &aggr = (LogicalAggregate &)op;
 			ClearUnusedExpressions(aggr.expressions, aggr.aggregate_index);
 
-			if (aggr.expressions.size() == 0 && aggr.groups.size() == 0) {
-				// removed all expressions from the aggregate: push a COUNT(*)
-				auto count_star_fun = CountStarFun::GetFunction();
-				aggr.expressions.push_back(
-				    AggregateFunction::BindAggregateFunction(context, count_star_fun, {}, false));
-			}
-		}
+            if (aggr.expressions.size() != 0 || aggr.groups.size() != 0) {
```

Comment:
The change here to an empty if followed by an else doesn't make much sense. Perhaps a left-over from earlier code?